### PR TITLE
adds a button to MapComponent to retrieve all nearby food places near atlanta

### LINKED ISSUE #61: Initial backend connection to Google Maps

### LINKED ISSUE #73: Figuring out Google Maps API Query
When looking for groceries stores and sources for healthy foods, what would our query be?

### PR DESCRIPTION
**Pull request by hannahli23**
Changes
----

- Added button to MapComponenet makes backend call
- Added props to GoogleMapComponenet for Lat and Long
- Added map.js which makes the api call to google maps and returns a JSON with all nearby stores

Screenshots
----
image
- 

Resolved issues
-----

Closes #61 
Closes #73

### CODE REVIEW

File: client/src/App.js [hunk 1]
```diff
@@ -30,19 +30,6 @@ function App() {
...
-                        return null;
-                    }
-                    if (!loading && !isLatestVersion) {
-                        // You can decide how and when you want to force reload
-                        refreshCacheAndReload();
-                    }
-                    return null;
-                }
-                }
-            </CacheBuster>
```
Comment (evan10s) [inline client/src/App.js]: Add this back please

File: server/app.js [hunk 2]
```diff
@@ -1,48 +1,54 @@
...
+    // set locals, only providing error in development
+    res.locals.message = err.message;
+    res.locals.error = req.app.get("env") === "development" ? err : {};
+
+    // render the error page
+    res.status(err.status || 500);
+    res.render("error");
+});
+
+const limiter = rateLimit({
```
Comment (evan10s) [inline server/app.js]: I think the comments aren't accurate here, right?  It looks like this is saying 5 requests in 15 minute period.

File: server/app.js [hunk 3]
```diff
@@ -1,48 +1,54 @@
...
+    windowMs: 1 * 60 * 1000, // 15 minutes
+    max: 5// limit each IP to 100 requests per windowMs
 });
 
+//  apply to all requests
+app.use(limiter);
```
Comment (evan10s) [inline server/app.js]: Shouldn't the rate limit only apply to API calls?

File: server/bin/www [hunk 4]
```diff
@@ -12,7 +12,7 @@ var http = require('http');
  * Get port from environment and store in Express.
  */
 
-var port = normalizePort(process.env.PORT || '3000');
+var port = normalizePort(process.env.PORT || '9000');
```
Comment (evan10s) [inline server/bin/www:15]: We should leave this at the default of 3000 IMO

File: server/routes/map.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
+const express = require("express");
+const router = express.Router();
+const axios = require("axios");
+require("dotenv").config;
+
+let key = process.env.GOOGLE_MAPS_API_KEY;
```
Comment (evan10s) [inline server/routes/map.js]: make this a `const`

File: server/routes/map.js [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+router.get("/query", async function(req, res, next) {
+    let lat = req.query.lat;
+    let long = req.query.long;
+    let radius = 50000;
+    let answer;
+    let results;
+    let PD = new placeDetails();
+
+    let host = "https://maps.googleapis.com";
+    let path = "/maps/api/place/nearbysearch/json?location=" + lat + "," + long
```
Comment (evan10s) [inline server/routes/map.js]: Use a template literal here instead of plain string concatenation, it's cleaner

File: package.json [hunk 7]
```diff
@@ -22,6 +22,7 @@
     "danger": "^10.0.0"
   },
   "dependencies": {
+    "cors": "^2.8.5",
```
Comment (evan10s) [inline package.json:25]: What is this CORS dependency for?

**Review by evan10s — CHANGES_REQUESTED** [review]
Exciting!

- Make sure you fix the Codacy style errors
- The continuous integration build is failing because the `yarn.lock` files need to be updated.  You can do this by running `yarn install` in the root of this repository and then committing the 1-3 resulting changed `yarn.lock` files.

File: server/bin/www (hunk 4 above)
Comment (hannahli23) [reply server/bin/www:15]: Hmm are you sure they can both run on port 3000

Comment (evan10s) [reply server/bin/www:15]: Well, they can't.  But the frontend and backend use separate development servers, and the proxy configuration on the frontend expects the server to be on port 3000.  Then you run the frontend with a PORT=3001 environment variable so there's no conflict.

File: server/routes/map.js [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+            url: total,
+            method: "get"
+        });
+        results = answer.data.results;
+    } catch(error) {
+        console.error(error);
+    }
+
+    PD.places.push(["name", "location", "place_id", "vicinity"])
+    for(let i = 0; i < results.length; i++) {
```
Comment (evan10s) [inline server/routes/map.js]:
```suggestion
    for (let i = 0; i < results.length; i++) {
```

File: package.json (hunk 7 above)
Comment (hannahli23) [reply package.json:25]: It's for the backend since it's at port 9000

**Comment by 9353oasis** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Files in the [`client`](/client) directory were modified; please update the client version.
  </td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>

  Files in the [`server`](/server) directory were modified; please update the server version.
  </td>
    </tr>
  </tbody>
</table>



<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>ℹ</td>
      <td>This pull request includes server dependency changes.</td>
    </tr>
  </tbody>
</table>


<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against d98ba69a2722bdaf2957b920a373b65e3f8a854e
</p>

**Comment by 9353oasis** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Files in the [`server`](/server) directory were modified; please update the server version.
  </td>
    </tr>
  </tbody>
</table>



<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>ℹ</td>
      <td>This pull request includes server dependency changes.</td>
    </tr>
  </tbody>
</table>


<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 14dca32c06b8e0e2fc9b4f1e15b1421d2590cc4d
</p>

**Comment by 9353oasis** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Files in the [`server`](/server) directory were modified; please update the server version.
  </td>
    </tr>
  </tbody>
</table>



<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>ℹ</td>
      <td>This pull request includes server dependency changes.</td>
    </tr>
  </tbody>
</table>


<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against a5e4d379afffcdb99223d0230e0f98ed905dbd5d
</p>

**Comment by 9353oasis** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>ℹ</td>
      <td>This pull request includes server dependency changes.</td>
    </tr>
  </tbody>
</table>


<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 5513a967479a7e6d7d6155332dcb4d146a6737c4
</p>

**Review by evan10s — APPROVED** [review]
Looks good, thanks!

**Pull request merged**